### PR TITLE
Begin work on adding cross platform support

### PR DESCRIPTION
**Pull request by kdawgwilk**
This is the beginnings to add cross platform support.  We recently started using LaunchDarkly and built a small wrapper around the client but then realized it only supported iOS so I figured I would give it a shot to support other Apple platforms

### CODE REVIEW

**Comment by kdawgwilk** [comment]
This is not ready to merge currently but I thought it might be good to get eyes on it asap

**Comment by ebriney** [comment]
@kdawgwilk : When do you think it will be ready? Because I have the same problem...

**Comment by arun251** [comment]
Hi @ebriney, thanks for the feedback. We will be updating and merging this PR this week.

File: Darkly/LDUserBuilder.m [hunk 1]
```diff
@@ -191,7 +191,11 @@ -(LDUserModel *)build {
         [user key:key];
         DEBUG_LOG(@"LDUserBuilder building User with key: %@", key);
     } else {
+#if TARGET_OS_OSX
+        NSString *uniqueKey = @"";
```
Comment (kdawgwilk) [inline Darkly/LDUserBuilder.m]: Not sure how to get a unique key on macOS

File: Podfile [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # platform :ios, '6.0'
 
 target 'Darkly' do
-  pod 'DarklyEventSource', :git => 'https://github.com/launchdarkly/ios-eventsource.git'
+    pod 'DarklyEventSource', :git => 'https://github.com/kdawgwilk/ios-eventsource.git', :branch => 'feature/add-support-for-tvos-watchos'
```
Comment (kdawgwilk) [inline Podfile]: This will need to be updated when the PR merges fixing the dependency

File: Pods/DarklyEventSource/EventSource/EventSource.m [hunk 3]
```diff
@@ -203,17 +186,17 @@ - (void)connection:(NSURLConnection *)connection didReceiveData:(NSData *)data
...
 
     if (wasClosed) {
         return;
     }
-    
+
     Event *e = [Event new];
     e.readyState = kEventStateClosed;
-    e.error = [NSError errorWithDomain:@""
+    e.error = error ?: [NSError errorWithDomain:@""
```
Comment (kdawgwilk) [inline Pods/DarklyEventSource/EventSource/EventSource.m]: Whoops not sure what happened here

**Comment by arun251** [comment]
Thanks for the PR @kdawgwilk . We have considerable work on another branch that we intend to merge soon. We will focus on that branch instead.

**Pull request closed without merging**